### PR TITLE
include option for nested dist

### PR DESCRIPTION
**Pull request by TomVasile**
Fixes issue where dist directory is assumed to be in the base directory. With this change, you can specify if that directory exists elsewhere.

Edit: Also fixed build dependencies since I was unable to build the project in it's current state without them.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ steps:
   pull: always
   image: golang:1.11-alpine
   commands:
-  - apk --no-cache add -U python3 git
+  - apk --no-cache add -U python3 git python3-dev gcc libc-dev libffi-dev openssl-dev
```
Comment (lafriks) [inline .drone.yml:22]: These changes don't seem to be related to this PR, just like changes below in Dockerfiles

Comment (TomVasile) [reply .drone.yml:22]: @lafriks You can't build the project without these changes currently.

Comment (TomVasile) [reply .drone.yml:22]: @lafriks @tboerger Please advise

**Comment by lafriks** [comment]
@tboerger could be more help in this as I'm not familiar with this plugin that much

**Comment by TomVasile** [comment]
Also I just noticed this, these dependency changes were approved already and were never merged.

https://github.com/drone-plugins/drone-pypi/pull/25

**Comment by lafriks** [comment]
Can you revise?

**Comment by TomVasile** [comment]
> Can you revise?

I can if you merge the other PR, the project does not build without those changes. Or close that PR and have them just merge in with mine. Whichever you guys prefer.

**Comment by TomVasile** [comment]
@lafriks Please let me know what you would like me to do here.

File: plugin.go [hunk 2]
```diff
@@ -40,7 +42,7 @@ func (p Plugin) uploadCommand() *exec.Cmd {
 	args = append(args, p.Username)
 	args = append(args, "--password")
 	args = append(args, p.Password)
-	args = append(args, "dist/*")
+	args = append(args, filepath.Join(filepath.Dir(p.DistDir), "/*"))
```
Comment (bradrydzewski) [inline plugin.go:45]: you can simplify to this:

```
filepath.Join(p.DistDir, "/*")
```

**Comment by tboerger** [comment]
Please rebase as the other PR have been merged.

File: plugin.go (hunk 2 above)
Comment (TomVasile) [reply plugin.go]: @bradrydzewski Thanks! Pushed this change.

**Comment by TomVasile** [comment]
> Please rebase as the other PR have been merged.

@tboerger Done

**Comment by bradrydzewski** [comment]
quick question ... wondering if we could simplify this a bit further?

```diff
		cli.StringFlag{
			Name:	"dist_dir",
			Usage:  "used when distribution directory is not in build root",
-			Value:  "dist/",
+			Value:  "dist/*",
			EnvVar: "PLUGIN_DIST_DIR",
		},
```

```diff
-args = append(args, filepath.Join(p.DistDir, "/*"))
+args = append(args, p.DistDir)
```

**Comment by TomVasile** [comment]
> quick question ... wondering if we could simplify this a bit further?
@bradrydzewski 
So the reason I did it that way was so that the end user didn't have to specify `dist/<files or *>` but only the path to which the dist directory is located.

I guess I could change it to be a "path_to_dist_file(s)" envvar.

What are your thoughts? I'm kind of indifferent as both fit my use case.

**Comment by bradrydzewski** [comment]
I think the below patch is preference since it is more consistent for how we handle this with other plugins.

```diff
		cli.StringFlag{
			Name:	"dist_dir",
			Usage:  "used when distribution directory is not in build root",
-			Value:  "dist/",
+			Value:  "dist/*",
			EnvVar: "PLUGIN_DIST_DIR",
		},
```

```diff
-args = append(args, filepath.Join(p.DistDir, "/*"))
+args = append(args, p.DistDir)
```

**Comment by TomVasile** [comment]
@bradrydzewski Sounds good, will pick this back up shortly.

File: plugin.go [hunk 3]
```diff
@@ -40,7 +42,7 @@ func (p Plugin) uploadCommand() *exec.Cmd {
 	args = append(args, p.Username)
 	args = append(args, "--password")
 	args = append(args, p.Password)
-	args = append(args, "dist/*")
+	args = append(args, filepath.Join(p.DistDir, "/*"))
```
Comment (tboerger) [inline plugin.go:45]: You can drop the slash, filepath join will add that.

Comment (tboerger) [reply plugin.go:45]: Scratch that, apply it like brad said.
